### ISSUE TITLE
Hashbrown/no_std support

### ISSUE THREAD

**Issue by MendyBerger** [jofas/map_macro#33]
I would love to have a feature flag that adds support for `hashbrown` data structures, so that I can use these macros with `no_std`.

**Comment by jofas** [comment]
Hi, thanks for the feature request. I've thought about adding `hashbrown` before but dismissed it as the `HashMap` and `HashSet` implementations from `std` are just wrappers around `hashbrown` anyway. What I didn't consider was `no_std` use-cases indeed. So adding macros for the `hashbrown` structures should be a great enhancement to `map-macro`!

**Comment by MendyBerger** [comment]
Can also just be a feature gate.

**Comment by jofas** [comment]
Yes, I will put them behind a feature gate, even though with macros it shouldn't really matter that much, given that they are expanded in the user code and not the library itself. I think it'd still impact build time given that it'd allow us to declare `hashbrown` as an [optional dependency](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies).

**Comment by MendyBerger** [comment]
I'd prefer not exporting it. Mostly because of versioning issues.

**Comment by jofas** [comment]
Yeah me too. I'll whip up something this weekend and release version `0.3.0` with macros for the `hashbrown` types.

**Comment by jofas** [comment]
Okay, just released `v0.3.0` with hashbrown support. You can now use something like

```rust
use map_macro::hashbrown::hash_map;

let hello = hash_map! {
    "en" => "Hello",
    "de" => "Hallo",
    "fr" => "Bonjour",
    "es" => "Hola",
    "cat" => "Hola",
    "🌍" => "👋",
};
```

to instantiate your hashbrown `HashMap`. `HashSet` is also supported. You'd need to enable the `hashbrown` feature in your `Cargo.toml` to get the macros though:

```toml
[dependencies]
map-macro = { version = "0.3", features = ["hashbrown"] }
```

**Comment by MendyBerger** [comment]
That was quick!
Thanks @jofas!
